### PR TITLE
Adding configurable border-radius

### LINKED ISSUE #1393: Allow components to change border radius through theming engine
### Description  
Currently e.g. the badge renders as a circle only and the toast always has 50% of its height as a border radius similar to other components with fixed border radius.

### Expected result  
We want to allow developers to customize the border radius for certain components according to the [Material Shape Values](https://material.io/design/shape/applying-shape-to-ui.html#baseline-shape-values).

### Components to implement border radius rounding
- [x] igx-badge 
- [x] igxButton 
- [x] igx-buttongroup 
- [x] igx-calendar 
- [x] igx-card 
- [x] igx-carousel 
- [x] igx-chip 
- [x] igx-dialog 
- [x] igx-drop-down 
- [x] igx-expansion-panel 
- [x] igx-input-group
- [x] igx-list
   - [x] igx-list-item
- [x] igx-navdrawer 
- [x] igx-snackbar 
- [x] igx-toast 
- [x] igxTooltip 
- [ ] igx-grid filter dialog

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #1393 

Additional information (check all that apply):

- [ ] Bug fix
- [x] New functionality
- [x] Documentation
- [ ] Demos
- [ ]  CI/CD

Checklist:

- [x]  All relevant tags have been applied to this PR
- [ ] This PR includes unit tests covering all the new code
- [ ] This PR includes API docs for newly added methods/properties
- [ ] This PR includes feature/README.MD updates for the feature docs
- [ ] This PR includes general feature table updates in the root README.MD
- [x] This PR includes CHANGELOG.MD updates for newly added functionality
- [x] This PR contains breaking changes
- [x] This PR includes ng update migrations for the breaking changes
- [ ] This PR includes behavioral changes and the feature specification has been updated with them

### Components list
- [x] igx-badge 
- [x] igxButton 
- [x] igx-buttongroup 
- [x] igx-calendar 
- [x] igx-card 
- [x] igx-carousel 
- [x] igx-chip 
- [x] igx-dialog 
- [x] igx-drop-down 
- [x] igx-expansion-panel 
- [x] igx-input-group
- [x] igx-list
   - [x] igx-list-item
- [x] igx-navdrawer 
- [x] igx-snackbar 
- [x] igx-toast 
- [x] igxTooltip

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/components/chip/_chip-theme.scss [hunk 1]
```diff
@@ -70,7 +70,12 @@
     $focus-selected-border-color: null
 ) {
     $name: 'igx-chip';
-    $theme: apply-palette(map-get($schema, $name), $palette);
+    $chip-schema: map-get($schema, $name);
+    $theme: apply-palette($chip-schema, $palette);
+
+    $border-radius: round-borders(
+        if($border-radius, $border-radius, map-get($chip-schema, 'border-radius')), 0, 20px
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/chip/_chip-theme.scss]: The material spec defines 16 as the maximum possible value.

File: projects/igniteui-angular/src/lib/core/styles/components/list/_list-theme.scss [hunk 2]
```diff
@@ -30,6 +34,8 @@
     $palette: $default-palette,
     $schema: $light-schema,
 
+    $list-border-radius: null,
+    $list-item-border-radius: null,
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/list/_list-theme.scss]: remove 'list' from the name of the params.

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_list.scss [hunk 3]
```diff
@@ -15,9 +15,16 @@
...
 ///
+/// @prop {Number} list-border-radius [0] - The border radius fraction, between 0-1 to be used for list component.
+/// @prop {Number} list-item-border-radius [0] - The border radius fraction, between 0-1 to be used for list item.
+///
 /// @see $default-palette
 
 $_light-list: (
+
+    list-border-radius: 0,
+    list-item-border-radius: 0,
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_list.scss]: Remove 'list' from the names of those properties.

File: projects/igniteui-angular/src/lib/core/styles/components/navdrawer/_navdrawer-theme.scss [hunk 4]
```diff
@@ -43,6 +45,8 @@
     $schema: $light-schema,
     $elevations: $elevations,
 
+    $right-border-radius: null,
+    $left-border-radius: null,
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/navdrawer/_navdrawer-theme.scss]: This should simply be one input parameter - '$border-radius'.

File: projects/igniteui-angular/src/lib/core/styles/components/snackbar/_snackbar-theme.scss [hunk 5]
```diff
@@ -59,6 +67,7 @@
     @return extend($theme, (
         name: $name,
         palette: $palette,
+        border-radiuslette: $border-radius,
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/snackbar/_snackbar-theme.scss]: Hm?

File: projects/igniteui-angular/src/lib/core/styles/components/snackbar/_snackbar-theme.scss [hunk 6]
```diff
@@ -95,6 +104,7 @@
         background: --var($theme, 'background');
         backface-visibility: hidden;
         box-shadow: --var($theme, 'shadow');
+        border-radius: --var($theme, 'border-radius')
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/snackbar/_snackbar-theme.scss]: Remove any previous border-radius declaration to fix linting errors thrown during build.

File: projects/igniteui-angular/src/lib/core/styles/components/toast/_toast-theme.scss [hunk 7]
```diff
@@ -28,11 +32,17 @@
...
     $background: null,
     $text-color: null
 ) {
     $name: 'igx-toast';
-    $theme: apply-palette(map-get($schema, $name), $palette);
+    $toast-schema: map-get($schema, $name);
+    $theme: apply-palette($toast-schema, $palette);
+
+    $border-radius: round-borders(
+        if($border-radius, $border-radius, map-get($toast-schema, 'border-radius')), 0, 24px
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/toast/_toast-theme.scss]: To keep the current design this should be between 0 - 26

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_toast.scss [hunk 8]
```diff
@@ -9,10 +9,13 @@
 ///
 /// @property {map} background [igx-color: ('grays', 600)] - The background color used for the toast.
 /// @property {map} text-color [igx-contrast-color: ('grays', 600)] - The text-color used for the toast.
+/// @prop {Number} border-radius [.17] - The border radius fraction, between 0-1 to be used for the toast component.
 ///
 /// @see $default-palette
 
 $_light-toast: (
+    border-radius: .17,
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_toast.scss]: To keep the current design, this should be 1.

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_tooltip.scss [hunk 9]
```diff
@@ -9,11 +9,13 @@
 ///
 /// @property {map} background [igx-color: ('grays', 700), hexrgba: (), rgba: .9] - The background color of the tooltip.
 /// @property {map} text-color [igx-color: ('grays', 700), hexrgba: (), rgba: .9, text-contrast: ()] - The text color of the tooltip.
-/// @property {Number} roundness [rem(4px)] - The border roundness of the tooltip. Smaller values mean less roundness.
+/// @prop {Number} border-radius [.1] - The border radius fraction, between 0-1 to be used for the tooltip component.
 ///
 /// @see $default-palette
 
 $_light-tooltip: (
+    border-radius: .1,
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_tooltip.scss]: .1 x 16 = 1.6, whereas the material spec states the default should be 4.

File: projects/igniteui-angular/src/lib/core/styles/components/list/_list-theme.scss [hunk 10]
```diff
@@ -17,7 +17,7 @@
 /// @param {Color} $item-text-color-active [null] - The active list item text color.
 ///
 /// @param {border-radius} $list-border-radius [null] - The border radius used for list component.
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/list/_list-theme.scss]: again, no need to use list in the name of the input param

File: projects/igniteui-angular/src/lib/core/styles/components/button/_button-theme.scss [hunk 11]
```diff
@@ -312,7 +312,7 @@
     $button-width: rem(88px);
     $button-height: rem(36px);
     $button-padding: 0 rem(16px, 16px);
-    $button-transition: all .15s ease-out;
+    $button-transition: background .15s ease-out, box-shadow .15s ease-out;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/button/_button-theme.scss]: The button group changes the text color on hover, so with this change the transition looks janky.

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/button/_button-theme.scss]: And why would you be changing the transition of the button in a pull request related to roundness?

Comment (desig9stein) [reply projects/igniteui-angular/src/lib/core/styles/components/button/_button-theme.scss]: Sorry it was a result of a merge conflict, I will fix it

File: projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_button.scss [hunk 12]
```diff
@@ -46,9 +46,21 @@
...
 $_light-button: (
-    button-roundness: 4px,
+
+    flat-border-radius: .2,
+
+    raised-border-radius: .2,
+
+    fab-border-radius: 1,
+
+    icon-border-radius: 1,
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/themes/schemas/light/_button.scss]: While this pull request on the back burner, the outlined button style was merged. We'll need outlined-border-radius for it, too.

**Pull request merged**